### PR TITLE
Fix type compatible issue for resource operation.

### PR DESCRIPTION
**Pull request by yaohaizh**
Signed-off-by: Yaohai Zheng <yaozheng@microsoft.com>

@mickaelistria  @spoenemann

### CODE REVIEW

**Comment by mickaelistria** [comment]
Wouldn't it be better to make it an enum with a dedicated type?

**Comment by yaohaizh** [comment]
For string value, we current already have CodeAction/MarkupKind related using this pattern. Just follow this pattern. 

@spoenemann, what's the specific reason at the beginning for this?

File: org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/Protocol.xtend [hunk 1]
```diff
@@ -67,13 +67,13 @@ class WorkspaceEditCapabilities {
 	 * The resource operations the client supports. Clients should at least
 	 * support 'create', 'rename' and 'delete' files and folders.
 	 */
-	List<ResourceOperationKind> resourceOperations;
+	List<String> resourceOperations;
```
[the remark this answers is not in the document]
Comment (yaohaizh) [reply org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/Protocol.xtend]: @spoenemann Updated the PR.

**Comment by spoenemann** [comment]
@mickaelistria let's discuss the enum issue with a broader perspective in #85.

**Comment by mickaelistria** [comment]
Thank you!

**Pull request merged**